### PR TITLE
Drop the node version 10 and add 16 for testing

### PR DESCRIPTION
**Pull request by dev-yakuza**
Currently, the `Node` version `12` is `Maintenance LTS`. So, we can drop the `Node` version `10`.

And add the `16` version to support it.

https://nodejs.org/en/about/releases/

Screen Shot 2021-08-30 at 11 08 58

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: ["10", "12", "14"]
+        node-version: ["12", "14"]
```
Comment (ybiquitous) [inline .github/workflows/test.yml:14]: Not add `16`?

```suggestion
        node-version: ["12", "14", "16"]
```

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "lint"
   ],
   "engines": {
-    "node": "^10.13.0 || >=12.0.0"
+    "node": ">=12.0.0"
```
Comment (ybiquitous) [inline package.json:16]: If supporting ESM, the following seems better:

```suggestion
    "node": "^12.20.0 || ^14.13.1 || >=16.0.0"
```

See https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#how-can-i-move-my-commonjs-project-to-esm

**Review by ybiquitous — COMMENTED** [review]
Thanks for creating this PR! I have some suggestions.

**Comment by dev-yakuza** [comment]
@ybiquitous 
Thanks!
I fixed them on https://github.com/sider/remark-preset-lint-sider/pull/84/commits/83a061bbb4870a7e63380d804c0f800397d81304

**Review by ybiquitous — APPROVED** [review]
Looks good! 👍🏼

**Comment by ybiquitous** [comment]
Please squash and merge. 🙏🏼

**Pull request merged**